plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/cwc/controller/ChatController.kt

```diff
@@ -116,13 +123,31 @@ class ChatController private constructor(
     }
 
     override suspend fun processPromptChatMessage(message: IncomingCwcMessage.ChatPrompt) {
+        var prompt = message.chatMessage
+        var queryResult: List<RelevantDocument> = emptyList()
+        val triggerId = UUID.randomUUID().toString()
+        var shouldAddIndexInProgressMessage: Boolean = false
+        if (prompt.contains("@workspace")) {
+            if (CodeWhispererSettings.getInstance().isProjectContextEnabled()) {
+                val projectContextController = ProjectContextController.getInstance(context.project)
+                prompt = prompt.replace("@workspace", "")
+                queryResult = projectContextController.query(prompt)
+                if (!projectContextController.getProjectContextIndexComplete()) shouldAddIndexInProgressMessage = true
+                logger.info("project context relevant document count: ${queryResult.size}")
+            } else {
+                sendOpenSettingsMessage(message.tabId)
+            }
+        }
+
         handleChat(
             tabId = message.tabId,
-            triggerId = UUID.randomUUID().toString(),
-            message = message.chatMessage,
+            triggerId = triggerId,
+            message = prompt,
             activeFileContext = contextExtractor.extractContextForTrigger(ExtractionTriggerType.ChatMessage),
             userIntent = intentRecognizer.getUserIntentFromPromptChatMessage(message.chatMessage),
             TriggerType.Click,
+            projectContextQueryResult = queryResult,
+            shouldAddIndexInProgressMessage = shouldAddIndexInProgressMessage
```

Comment:
does `query()` return an empty list if indexing is still in progress?

Reply:
not necessarily, it can still return partial result if some files are indexed